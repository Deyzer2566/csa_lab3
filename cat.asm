b =_start
b =_int_handler
out_register: word 0xfffffc
in_register: word 0xfffffd
_start:
    b =_start
_int_handler:
    push r0
    push r1
    push r2

    ldr r0, [=in_register]
    ldr r1, [=out_register]
    ldr r2, [r0]
    cmp r2, 0
    haltz
    str r2, [r1]
    or r2, -1
    str r2, [r0]

    pop r2
    pop r1
    pop r0
    exint