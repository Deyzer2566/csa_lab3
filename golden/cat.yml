in_source: |-
 b =_start
 b =_int_handler
 out_register: word 0xfffffc
 in_register: word 0xfffffd
 _start:
     b =_start
 _int_handler:
     push r0
     push r1
     push r2

     ldr r0, [=in_register]
     ldr r1, [=out_register]
     ldr r2, [r0]
     cmp r2, 0
     haltz
     str r2, [r1]
     or r2, -1
     str r2, [r0]

     pop r2
     pop r1
     pop r0
     exint
in_stdin: |-
  foo
out_code: |-
  b'0\x00\x00\x0c0\x00\x00\x0c\x00\xff\xff\xfc\x00\xff\xff\xfd0\x00\xff\xfc \x00\x00\x00 \x02\x00\x00 \x04\x00\x00\x10\x00\xff\xe8\x10\x02\xff\xe0\x10\x05\x00\x00\xb0\x04\x00\x00\xa1\x80\x00\x00\x18\x05\x10\x00X\x04\xff\xff\x18\x05\x00\x00(\x04\x00\x00(\x02\x00\x00(\x00\x00\x00\xb8\x00\x00\x00'
out_stdout: |
  fooВыполнено тактов: 1566
